processor/processor.go

```diff
@@ -179,19 +194,143 @@ func (proc *HandleT) Setup(backendConfig backendconfig.BackendConfig, gatewayDB
 	}
 }
 
+type StoreErrorOutputT struct {
+	Location string
+	Error    error
+}
+
+func (proc *HandleT) storeErorrsToObjectStorage(jobs []*jobsdb.JobT) StoreErrorOutputT {
+	localTmpDirName := "/rudder-proc-err-logs/"
+
+	uuid := uuid.NewV4()
+	logger.Debug("[Processor: storeErorrsToObjectStorage]: Starting logging to object storage")
+
+	tmpDirPath, err := misc.CreateTMPDIR()
+	if err != nil {
+		panic(err)
+	}
+	path := fmt.Sprintf("%v%v.json", tmpDirPath+localTmpDirName, fmt.Sprintf("%v.%v.%v.%v", time.Now().Unix(), config.GetEnv("INSTANCE_ID", "1"), fmt.Sprintf("%v-%v", jobs[0].JobID, jobs[len(jobs)-1].JobID), uuid))
+
+	gzipFilePath := fmt.Sprintf(`%v.gz`, path)
+	err = os.MkdirAll(filepath.Dir(gzipFilePath), os.ModePerm)
+	if err != nil {
+		panic(err)
+	}
+	gzWriter, err := misc.CreateGZ(gzipFilePath)
+	if err != nil {
+		panic(err)
+	}
+	defer os.Remove(gzipFilePath)
+
+	var contentSlice [][]byte
+	for _, job := range jobs {
+		rawJob, err := json.Marshal(job)
+		if err != nil {
+			panic(err)
+		}
+		contentSlice = append(contentSlice, rawJob)
+	}
+	content := bytes.Join(contentSlice[:], []byte("\n"))
+	gzWriter.Write(content)
+	gzWriter.CloseGZ()
+
+	outputFile, err := os.Open(gzipFilePath)
+	if err != nil {
+		panic(err)
+	}
+	prefixes := []string{"rudder-proc-err-logs", time.Now().Format("01-02-2006")}
```

Comment:
Since this is visible to the customers, shall we add a more readable name? `processor-errors`

Reply:
changed